### PR TITLE
fix: trigger aligned timerule when diff is equal to period

### PR DESCRIPTION
**Pull request by ivaaaan**
Aligned time rule is not triggered on timestamps like `15:00:00` with one minute rule

### CODE REVIEW

File: src/aggregation_rules/aligned_time_rule.rs [hunk 1]
```diff
@@ -96,4 +97,43 @@ mod tests {
...
+            Trade {
+                timestamp: 1712656815000,
+                price: 101.0,
+                size: -10.0,
+            },
+            Trade {
+                timestamp: 1712656860000,
+                price: 101.0,
+                size: -10.0,
+            },
```
Comment (MathisWellmann) [inline src/aggregation_rules/aligned_time_rule.rs]:
```suggestion
            Trade {
                timestamp: 1712656815000,
                price: 100.5,
                size: 10.0,
            },
            Trade {
                timestamp: 1712656860000,
                price: 101.0,
                size: -10.0,
            },
```
Would be good for both prices to be unique.

File: src/aggregation_rules/aligned_time_rule.rs [hunk 2]
```diff
@@ -96,4 +97,43 @@ mod tests {
...
+        ];
+
+        let mut aggregator = GenericAggregator::<OhlcCandle, AlignedTimeRule, Trade>::new(
+            AlignedTimeRule::new(M1, TimestampResolution::Millisecond),
+        );
+        let candles = aggregate_all_trades(&trades, &mut aggregator);
+        assert_eq!(candles.len(), 2);
+        assert_eq!(candles[0].open(), 100.00);
+        assert_eq!(candles[1].open(), 101.00);
```
Comment (MathisWellmann) [inline src/aggregation_rules/aligned_time_rule.rs]:
```suggestion
        assert_eq!(candles[0].open(), 100.0);
        assert_eq!(candles[0].close(), 100.5);
        assert_eq!(candles[1].open(), 101.0);
        assert_eq!(candles[1].close(), 101.0);
```
An assertion for the close price would be good as well.
I think those values are correct, but unfortunate that both open and close prices are 101, so maybe it makes sense to add another trade to be included in the second candle

**Review by MathisWellmann — CHANGES_REQUESTED** [review]
Looks good overall. Some slight test modifications and we can get this merged.

**Comment by ivaaaan** [comment]
@MathisWellmann updated this one as well

**Review by MathisWellmann — APPROVED** [review]

**Pull request merged**